lib/modules/datasource/aws-lambda-layer/index.spec.ts

```diff
@@ -0,0 +1,329 @@
+        'xy',
+        'arm64',
+        'python3.8',
+      );
+
+      expect(res).toEqual([]);
+    });
+
+    it('should return array with one layer if one layer found', async () => {
+      mockListLayerVersionsCommandOutput(mock1Layer);
+      const lamdbaLayerDatasource = new AwsLambdaLayerDataSource();
+      const res = await lamdbaLayerDatasource.getSortedLambdaLayerVersions(
+        'not-relevant',
+        'not-relevant',
+        'not-relevant',
+      );
+
+      expect(res).toEqual([layer3]);
+    });
+
+    it('should return array with three layers if three layers found', async () => {
+      mockListLayerVersionsCommandOutput(mock3Layers);
+      const lamdbaLayerDatasource = new AwsLambdaLayerDataSource();
+      const res = await lamdbaLayerDatasource.getSortedLambdaLayerVersions(
+        'not-relevant',
+        'not-relevant',
+        'not-relevant',
+      );
+
+      expect(res).toEqual([layer1, layer2, layer3]);
+    });
+
+    it('should pass the filters for listLayerVersions when calling the AWS API', async () => {
+      mockListLayerVersionsCommandOutput(mock3Layers);
+      const lambdaLayerDatasource = new AwsLambdaLayerDataSource();
+
+      await lambdaLayerDatasource.getSortedLambdaLayerVersions(
+        'arn',
+        'runtime',
+        'architecture',
+      );
+
+      expect(lambdaClientMock.calls()).toHaveLength(1);
+
+      expect(lambdaClientMock.calls()[0].args[0].input).toEqual({
+        CompatibleArchitecture: 'architecture',
+        CompatibleRuntime: 'runtime',
+        LayerName: 'arn',
+      });
+    });
+
+    it('should call AWS with no architecture if there is no architecture in the filter', async () => {
+      mockListLayerVersionsCommandOutput(mock3Layers);
+      const lambdaLayerDatasource = new AwsLambdaLayerDataSource();
+
+      await lambdaLayerDatasource.getSortedLambdaLayerVersions(
+        'arn',
+        'runtime',
+        undefined,
+      );
+
+      expect(lambdaClientMock.calls()).toHaveLength(1);
+
+      expect(lambdaClientMock.calls()[0].args[0].input).toEqual({
+        CompatibleArchitecture: undefined,
+        CompatibleRuntime: 'runtime',
+        LayerName: 'arn',
+      });
+    });
+
+    it('should call AWS with no runtime if there is no runtime in the filter', async () => {
+      mockListLayerVersionsCommandOutput(mock3Layers);
+      const lambdaLayerDatasource = new AwsLambdaLayerDataSource();
+
+      await lambdaLayerDatasource.getSortedLambdaLayerVersions(
+        'arn',
+        undefined,
+        'python3.8',
+      );
+
+      expect(lambdaClientMock.calls()).toHaveLength(1);
+
+      expect(lambdaClientMock.calls()[0].args[0].input).toEqual({
+        CompatibleArchitecture: 'python3.8',
+        CompatibleRuntime: undefined,
+        LayerName: 'arn',
+      });
+    });
+  });
+
+  describe('getReleases', () => {
+    it('should throw an exception if the filter criteria does not match the schema', async () => {
```

Comment:
@viceice Is this assumption correct? If not: how to catch this error?

```
try {
  FilterParser.parse(xxxx);
} catch (e) {
  // log error and return null
}
```

Looks ugly and I guess there is a better option. Checking the existing source, I didn't find any.